mapping/collection.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+module Mapping
+  class AdminSet
+    require 'csv'
+    require 'yaml'
+
+    extend Extensions::BasicValueHandler
+
+    ##
+    # Convert data in a csv file to hash in YAML format
+    #
+    # @param[String] value - the path to the csv file
+    # return - [Hash] of YAML
+    def csv_to_yaml(value)
+      data = CSV.read (value, :headers => true).map(&:to_hash)
+      data.to_yaml
+    end
+
+  ##
+  # Get ownerId from object_properties and parse DSpace owning collection name from lookup table 
+  def owner_collection_name
+    lookup = csv_to_yaml('../../lookup/collectionlist.csv')
+    collection_list = lookup.select { |l| l[:collection_handle] == object_properties['ownerId'] }
+    [ collection_list[:collection_name] ]
```

Comment:
Since I believe we decided that the `owner_collection_name` should return only a single name, you would do something like how the AdminSet lookup works;

```
# find the first row where collection_handle matches
row = lookup.find { |l| l[:collection_handle] == value } 
row[:collection_name]
```